calcom/cal.com
# fix: allow gCal installs without a primary calendar

## What does this PR do?

This PR enhances the Google Calendar integration by improving the handling of primary calendars. It addresses scenarios where a user's primary calendar might not be set or available. The changes include:

- Fallback to the first available calendar if no primary calendar is found
- Skip creating a selected calendar if no suitable calendar is available
- Redirect the user to the appropriate page even if calendar creation is skipped

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

1. Set up a Google Calendar integration with Cal.com
2. Test with a Google account that has no primary calendar set
3. Verify that the integration falls back to the first available calendar
4. Test with a Google account that has no calendars at all
5. Confirm that the user is redirected properly without errors
6. Check that the selected calendar can be toggled on later in the app settings

Expected outcome:
- The integration should work smoothly for users with or without a primary calendar
- No errors should occur when handling accounts with no calendars
- Users should be able to manage their calendar selection post-integration

## Checklist

- I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code follows the style guidelines of this project
- I have commented my code, particularly in hard-to-understand areas
- I have checked if my changes generate no new warnings
